javascriptlib/src/mill/javascriptlib/ReactScriptsModule.scala

```diff
@@ -0,0 +1,174 @@
+package mill.javascriptlib
+
+import mill._
+import os.*
+
+// create-react-app scripts
+trait ReactScriptsModule extends TypeScriptModule {
+  def npmDevDeps: T[Seq[String]] = Task { Seq.empty[String] }
+
+  override def npmInstall: T[PathRef] = Task.Anon {
+    os.call((
+      "npm",
+      "install",
+      "--save-dev",
+      "react@18.3.1",
+      "react-dom@18.3.1",
+      "react-scripts@5.0.1",
+      "typescript@^4.9.5",
+      "web-vitals@2.1.4",
+      "@testing-library/jest-dom@^5.17.0",
+      "@testing-library/react@^13.4.0",
+      "@testing-library/user-event@^13.5.0",
+      "@types/jest@^27.5.2",
+      "@types/node@^16.18.119",
+      "@types/react@^18.3.12",
+      "@types/react-dom@^18.3.1",
+      "serve@12.0.1",
+      npmDevDeps(),
+      transitiveNpmDeps()
+    ))
+    PathRef(Task.dest)
+  }
+
+  override def sources: Target[PathRef] = Task.Source(millSourcePath)
+
+  def setup: Task[Unit] = Task.Anon {
+    // copy src files
+    os.copy(sources().path, Task.dest, mergeFolders = true)
+
+    def buildDependencies(dependencies: Seq[String]): ujson.Obj = {
+      val dependenciesObj = dependencies.map { dep =>
+        val lastAtIndex = dep.lastIndexOf('@')
+        if (lastAtIndex == -1)
+          dep -> ujson.Str("latest") // No '@' found, default version to "latest"; you probably should always provide the dependency version
+        else {
+          val name = dep.substring(0, lastAtIndex)
+          val version = dep.substring(lastAtIndex + 1)
+          name -> ujson.Str(version)
+        }
+      }
+      ujson.Obj.from(dependenciesObj)
+    }
+
+    val allPaths = Seq(
+      "*" -> Task.dest / "node_modules",
+      "app/*" -> "app",
+      "typescript" -> Task.dest / "node_modules" / "typescript"
+    )
+
+    // mk tsconfig.json
```

Comment:
Is this format documented anywhere? If so we should link it in a comment

Reply:
yeah this is create-react-app pretty standard. I will add a link